code/obj/flock/cage.dm

```diff
@@ -164,8 +164,12 @@
 	process()
 		// process fluids into stuff
 		if(reagents.has_reagent(target_fluid, create_egg_at_fluid))
-			reagents.remove_reagent(target_fluid, create_egg_at_fluid)
-			spawnEgg()
+			if (src.flock?.getComplexDroneCount() < FLOCK_DRONE_LIMIT)
+				spawnEgg()
+			else
+				src.reagents.reaction(get_turf(src), react_volume = create_egg_at_fluid/2)
```

Comment:
maybe resource cubes instead of flock juice?

Reply:
Yeah resources cubes would be cool